### DISCUSSION TITLE
dragging the map view using R-MouseButton and MouseMove

### DISCUSSION THREAD

**Discussion by nnako** [freeplane/freeplane#388] [Ideas]
Currently, in order to scroll the entire map, a user will use (at least on Windows systems) these controls:

`<L-CLICK-AND-HOLD>` on an empty position within the map and then
`<MOUSEMOVE>` until the map is centered at the desired location

Some other applications (e.g. yED blockdiagram editor) use something different:

`<R-CLICK-AND-HOLD>` on any position within the map (regardless any nodes location)
`<MOUSEMOVE>` ...

Using the latter operation style, the `<L-CLICK>` could consequently be used for **selecting nodes** within the map, rather than for selecting nodes _and_ moving the map. When using `<R-CLICK>` for dragging the map in the described way, there could be some benefits for using `<L-CLICK>` in the future:

- `<L-CLICK>` on single or multiple nodes (using `<SHIFT>` or `<CTRL>`) to select nodes for application of actions in a subsequent operation step (this is current use)
- `<L-CLICK-AND-HOLD>` on a selected (or a selected group of) node and `<MOUSEMOVE>` to move the selected nodes to another parent (this is current use)
- `<L-CLICK-AND-HOLD>` on an empty spot on the map, and `<MOUSEMOVE>` to show a rubber-band "selection box". Any node within this box would be selected / added for further copy, delete, move to another parent, ...

This alternative selection behavior could even be activated via settings. For users who rather like to stick with the current use of `<L-CLICK>` and `<R-CLICK>`.

What do you think?

**Comment by euu2021** [comment]
When I was alternating a lot between Freeplane and Yed, it was annoying having to use different pan/select keys for each one. It´s not easy to make the mind change gears quickly. So, having the option to choose is useful for people who need to use multiple programs.

There is also the Spacebar+leftClick to pan used by adobe.

**Reply by nnako** [reply]
> It´s not easy to make the mind change gears quickly. So, having the option to choose is useful for people who need to use multiple programs.

absolutely. having to think about the keys to be pressed within a particular tool, all the time, significantly reduces the focus and creative abilities in that moment.

And, as we will most probably not be successful in convincing / persuading yWorks to change the user interface of yED, we might create options to render Freeplane even more flexible as it already is.
